File: source_funcs/go_transcoder/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL.go
package main

func f_gold(str []byte) int {
	var checker int = 0
	for i := int(0); i < int(len(str)); i++ {
		var val int = int(str[i] - byte('a'))
		if (checker & (1 << val)) > 0 {
			return i
		}
		checker |= 1 << val
	}
	return -1
}

File: rust_funcs/cpp_transcoder/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL/EFFICIENTLY_FIND_FIRST_REPEATED_CHARACTER_STRING_WITHOUT_USING_ADDITIONAL_DATA_STRUCTURE_ONE_TRAVERSAL.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::os::raw::c_char;

fn f_gold(str: String) -> i32 {
    let mut checker: i32 = 0;
    for i in 0..str.len() {
        let val: i32 = str.chars().nth(i).unwrap() as i32 - 'a' as i32;
        if checker & (1 << val) > 0 {
            return i as i32
        }
        checker |= 1 << val;
    }
    -1
}
